packages/render-three/src/CSG2Three.js

```diff
@@ -0,0 +1,88 @@
+export function CSG2Three (THREE) {
+  const {
+    MeshPhongMaterial,
+    LineBasicMaterial,
+    BufferGeometry,
+    BufferAttribute,
+    Mesh,
+    InstancedMesh,
+    Line,
+    LineSegments,
+    Color
+  } = THREE
+
+  const flatShading = true
+  const materials = {
+    mesh: {
+      def: new MeshPhongMaterial({ color: 0x0084d1, flatShading }),
+      make: params => new MeshPhongMaterial({ flatShading, ...params })
+    },
+    line: {
+      def: new LineBasicMaterial({ color: 0x0000ff }),
+      make: params =>
+        new LineBasicMaterial(params)
+    },
+    lines: null
+  }
+  materials.lines = materials.line
+
+  function _CSG2Three (obj) {
+    const { vertices, indices, normals, color, colors, isTransparent = false, opacity } = obj
+    let { transforms } = obj
+    const objType = obj.type || 'mesh'
+
+    const materialDef = materials[objType]
+    if (!materialDef) { console.error('material not found for type ', objType, obj) }
+    let material = materialDef.def
+    const isInstanced = obj.type === 'instances'
+    if ((color || colors) && !isInstanced) {
+      const c = color || colors
+      const opts = {
+        vertexColors: !!colors,
+        opacity: c[3] === undefined ? 1 : c[3],
+        transparent: (color && c[3] !== 1 && c[3] !== undefined) || isTransparent
+      }
+      if (opacity) opts.opacity = opacity
```

Comment:
Wouldn't be better an undefined or just set a default value for opacity as well?

Reply:
I think here is more issue of handling user input, I need to check if array is rgb (c[3] === undefined) or it is rgba but opacity is full c[3] === 1

this handling of opacity in general for the 3 engines is something to look at at later point during imrpovement of code.
I would leave it as is for this PR